### ISSUE TITLE
Airtap\Select not working

### ISSUE THREAD

**Issue by krishnan-unni** [Microsoft/MixedRealityToolkit-Unity#3428]
## Overview
Airtap gesture is not getting fired

## Expected Behavior
Airtap gesture should get fired on press of Space or Enter in emulator

## Actual Behavior
Airtap gesture is not getting fired on press of Space or Enter in emulator

## Steps to reproduce
Create a new project.,
Setup profiles from scratch. Create Action profile and add an action called Select with Digital Axis. Create Grip action with 6 DoF
Create Controller profile and select Hololens.
Assign the Select action to Air tap, Assign None to Spatial pointer, Grip action to Spatial Grip
Create a Cube and add a Cube controller. Add a simple function OnTap to handle tap event.
Add Pointer Click Handler script to the Cube and hook the OnTap function to the select Action.
In the MRTKConfigurationProfile set the InputSystem type to MixedRealityInputSystem
Run in Unity or Emulator. The Air tap gesture never gets triggered.

## Unity Editor Version
2018.3.3f1 - 5f1

## Mixed Reality Toolkit Release Version
mrtk_development

![1](https://user-images.githubusercontent.com/14880339/52609843-a9a3fe80-2e4c-11e9-809e-bed697381873.jpg)

![mrtk102](https://user-images.githubusercontent.com/14880339/52609847-add01c00-2e4c-11e9-8502-21922bc646b4.jpg)

![2](https://user-images.githubusercontent.com/14880339/52609848-af99df80-2e4c-11e9-9d97-765fa670f72a.jpg)

**Comment by krishnan-unni** [comment]
Is this related to #3442 ?

**Comment by krishnan-unni** [comment]
Any help on this please?

**Comment by keveleigh** [comment]
Strange! I haven't tested in the emulator. @davidkline-ms, do you have any insight into this?

**Comment by krishnan-unni** [comment]
It's not working in the Unity Editor Game view either. I thought Point Click Handler is platform independent generic event. That is what I see after looking at the code.

**Comment by chbecker-ms** [comment]
@krishnan-unni - We chatted about #3428 and #3442 at our weekly shiproom. The main issue is getting a repro locally.

Are you on the slack channel? We'd love to start a conversation: https://holodevelopers.slack.com/messages/C2H4HT858

**Comment by krishnan-unni** [comment]
Hi I am in the channel now. We can talk

**Comment by davidkline-ms** [comment]
> Strange! I haven't tested in the emulator. @davidkline-ms, do you have any insight into this?

I don't at this time. We should definitely investigate asap.

**Comment by davidkline-ms** [comment]
Possibly related to #3456. We will continue to investigate and advise in both issues.

**Comment by Alreschat** [comment]
I'm having the same issue on unity 2018.3.6f1

**Comment by keveleigh** [comment]
Which emulator versions are people using?

**Comment by krishnan-unni** [comment]
The one I am using is 10.0.17134.80. Thats the latest release build

**Comment by keveleigh** [comment]
Hmm, I'm not able to repro this in either the 10.0.17134.80 emulator or the 10.0.17763.253 emulator. It might be worth trying the 17763 emulator, which can be found at https://docs.microsoft.com/en-us/windows/mixed-reality/hololens-emulator-archive
![image](https://user-images.githubusercontent.com/3580640/53192814-332d9c00-35c4-11e9-93ea-c07b7f3d7d52.png)


I am able to repro it using Simulate in Editor inside the Unity Editor, as additionally reported by #3456:
![image](https://user-images.githubusercontent.com/3580640/53192620-cb775100-35c3-11e9-8913-43c815abedc8.png)

**Comment by krishnan-unni** [comment]
@keveleigh Im gonna try in the latest build of emulator. Will update you ASAP. In the mean time could you please share a screen of your Hololens controller and action profile mapping too? Just want to make sure it is not a config issue

**Comment by krishnan-unni** [comment]
I just tried this with Unity 2018.3.6f1, Hololens emulator 10.0.17763.134. Updated the mrtk_development branch. Still the Air Tap is not working. I have committed it to my git repository here: [https://github.com/krishnan-unni/MRTK-101](url) if you would like to check it.

**Comment by keveleigh** [comment]
@krishnan-unni Can you check under Window -> Package Manager and see which version of the Windows Mixed Reality package you have is please? I believe there was an issue with 1.0.7 that should be fixed in 1.0.8.

**Comment by keveleigh** [comment]
Also, my profile mappings are:

**Comment by krishnan-unni** [comment]
@keveleigh  Mine is 1.0.8 too. The Solution you downloaded from my git. Is the Tap working when tried in your system? Im running Unity 2018.3.6f1 now. and Upgraded the emulator too. The mapping looks fine. (Tap is a Digital action and a 6DoF action for the other two)

**Comment by krishnan-unni** [comment]
@davidkline-ms @keveleigh Finally, I figured it out. Not sure if this is a bug or a config requirement. In the Input System Profile I had set Controller Visualization profile as None since it is for hololens, I don't have a controller to render.  Although this was why Air Tap was not triggering. Once I placed the Default or a newly created Controller Visualization profile it started working. This one really sent me on a goose chase with days of code debug and trial and error attempts.

![1](https://user-images.githubusercontent.com/14880339/53292651-97eb2100-3793-11e9-95ee-0ce255b0ffa1.jpg)

![2](https://user-images.githubusercontent.com/14880339/53292655-b7824980-3793-11e9-9a05-e7f12a4a6509.jpg)

So the question is, is this a bug (sounds to me that way) or a configuration requirement. If it is a config requirement it would be better to clarify to the user in the documentation and why. I'm still combing through the code to see what is the connection between this profile and Input Click. In the mean time please let me know if I need to raise a new issue in case this is a bug.

**Comment by davidkline-ms** [comment]
We believe the v2 beta 2 release has addressed this issue when using remoting and holographic simulation. 

@krishnan-unni, if you believe you have found an issue please feel free to open a new bug. We are more than happy to help investigate and review.

Thanks!

**Comment by davidkline-ms** [comment]
If you are still seeing this issue, please reactivate and we will investigate further.

Thanks!

**Comment by krishnan-unni** [comment]
Hi @davidkline-ms  I tried with both mrtk_development and the latest Beta 2.0.0 v2 unity package from here: [https://github.com/Microsoft/MixedRealityToolkit-Unity/releases/tag/v2.0.0-Beta2](url)

I was able to reproduce it in both. I have the project uploaded here: [https://github.com/krishnan-unni/MRTK-101](url) 
If you simply set the Visual Controller Profile to None in the Input System Profile you can reproduce it. Once you set that to the Default profile or a new one it will work fine again.

**Comment by krishnan-unni** [comment]
So you want me to open a new bug or reactivate this?

**Comment by davidkline-ms** [comment]
Please open a new bug with the specific repro steps. Thanks!

**Comment by Avalin** [comment]
I have the same setup and same issue, **however** I attempted testing it inside Unity itself and on a real HoloLens (not emulator). It actually works fine when testing in the Unity editor, the cube gets selected. When it is built on HoloLens however, it does not respond to any tapping gestures. I have a similar setup as OP. I followed the guide for this setup [found here](http://unicodeexception.com/2019/02/start-mixedrealitytoolkit-v2/). Did I miss something?
